0008896 - :sparkles: Afficher la corbeille des documents de l'espace

Rotomeca opened the pull request:
Comme on affiche "Documents" et "Favoris" dans le widget Documents de l'espace de travail, ajouter un 3eme menu "Corbeille" qui affiche les fichiers supprimés dans l'espace de travail

+ Documentation des fichiers liés

Rotomeca committed:
feat(nextcloud/workspace): :sparkles: Afficher la corbeille des documents de l'espace

Comme on affiche "Documents" et "Favoris" dans le widget Documents de l'espace de travail, ajouter un 3eme menu "Corbeille" qui affiche les fichiers supprimés dans l'espace de travail

FEAT: MANTIS 0008896

tpayen commented on plugins/roundrive/lib/BnumWebDavAdaptater.php:
```diff
@@ -45,4 +102,30 @@ public function listFavorites() : \Generator
+
+        foreach ($object as $key => $value) {
+            switch ($key) {
+                case '{http://nextcloud.org/ns}trashbin-title':
```
J'ai l'impression que ce qui est affiché se base sur le le `href` alors qu'il serait pas mal de se baser sur `trashbin-filename` par exemple pour pas avoir un nom de sauvegarder derrière (ou je ne sais pas trop quoi)

exemple :
```
<d:response><d:href>/mdrive/remote.php/dav/trashbin/thomas.payen/trash/test.md.d1736497624</d:href><d:propstat><d:prop><nc:trashbin-filename>test.md</nc:trashbin-filename><nc:trashbin-deletion-time>1736845406</nc:trashbin-deletion-time><nc:trashbin-original-location>test.md</nc:trashbin-original-location><nc:trashbin-title>dossiers-dev-du-bnum-1/test.md</nc:trashbin-title><d:getlastmodified>Fri, 10 Jan 2025 08:27:04 GMT</d:getlastmodified><d:getetag>6780d9e0a941c</d:getetag><d:getcontenttype>text/markdown</d:getcontenttype><d:resourcetype/><oc:fileid>65453709</oc:fileid><oc:permissions>GD</oc:permissions><oc:size>0</oc:size><d:getcontentlength>0</d:getcontentlength>
``` 

![image](https://github.com/user-attachments/assets/e762ed8c-de09-4279-9533-b74bb7003ccc)

tpayen commented on plugins/mel_nextcloud/skins/elastic/templates/workspace/block.html:
```diff
@@ -4,6 +4,7 @@
             <bnum-voice>Choisissez une vue</bnum-voice>
             <selected-option value="default">Documents</selected-option>
             <bnum-option value="favorites">Favoris</bnum-option>
+            <bnum-option value="trash">Corbeille</bnum-option>
```
J'étais passé a côté pour les favoris, mais ça mériterait une petite localization ici

Rotomeca replied:
Ah oui, j'ai zapé aussi

Rotomeca committed:
style(nextcloud/workspace): :globe_with_meridians: Ajout des localizations

Il manquait des localisations pour le html lié au module nexctloud d'un edt.

Rotomeca committed:
fix(nextcloud/workspace): :bug: Le nom des fichiers n'étaient pas bon.

On se basait sur la prop' `path` qui n'était pas mis à jours pour avoir le bon chemin.

Rotomeca committed:
refactor(nextcloud/workspace): :pencil2: Il y avait une faute de le nom d'une fonction

`Tashes` au lieu de `Trashes`.
